src/basestation/session.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef SESSION_H
+#define SESSION_H
```

Comment:
Switch to `#pragma once` instead of include guards: this is a code standard change I didn't mention (it was previously unstandardized on our team). Virtually all compilers on every platform support `#pragma once`, and it's just a bit cleaner.